src/main/scala/firrtl/stage/FirrtlAnnotations.scala

```diff
@@ -64,6 +64,41 @@ object FirrtlFileAnnotation extends HasShellOptions {
 
 }
 
+/** Read a directory of FIRRTL files or ProtoBufs
+  *  - set with `-I/--input-directory`
+  * @param file input filename
+  */
+case class FirrtlDirectoryAnnotation(dir: String) extends NoTargetAnnotation with CircuitOption {
+
+  def toCircuit(info: Parser.InfoMode): FirrtlCircuitAnnotation = {
+    val circuit =
+      try {
+        proto.FromProto.fromDirectory(dir)
+      } catch {
+        case a @ (_: FileNotFoundException | _: NoSuchFileException) =>
+          throw new OptionsException(s"Directory '$dir' not found! (Did you misspell it?)", a)
+        case _: NotDirectoryException =>
+          throw new OptionsException(s"Directory '$dir' is not a directory")
+      }
+    FirrtlCircuitAnnotation(circuit)
+  }
+
+}
+
+object FirrtlDirectoryAnnotation extends HasShellOptions {
+
+  val options = Seq(
+    new ShellOption[String](
+      longOption = "input-directory",
+      toAnnotationSeq = a => Seq(FirrtlDirectoryAnnotation(a)),
+      helpText = "A directory of FIRRTL files",
```

Comment:
can these be either `.fir` or `.pb` files? A mixture?

Reply:
We should figure that out. I think to start we should make it be only `.fir` or only `.pb` files. Note that `-i` actually doesn't care about file extensions, it just tries to deserialize as protobuf and if it fails it deserializes as `.fir` but I think it's okay to require correct file extensions here.